Source/CMakeLists.txt

```diff
@@ -18,11 +18,21 @@ if(CMAKE_C_COMPILER_ID MATCHES "MSVC")
 else()
   # Enable C++17, but fall back to C++14 if it isn't available.
   # CMAKE_CXX_STANDARD cannot be used here because we require C++14 or newer, not any standard.
-  check_and_add_flag(CXX17 -std=c++17)
-  if(NOT FLAG_CXX_CXX17)
-    set(CMAKE_CXX_STANDARD 14)
-    set(CMAKE_CXX_STANDARD_REQUIRED ON)
-    set(CMAKE_CXX_EXTENSIONS OFF)
+  # On Android, it is necessary to use C++1z instead, since the C++17 support is currently incomplete
+  if (NOT ANDROID)
```

Comment:
This should probably be inverted (Android case coming first), so what the comment is describing is immediately below it